### PR TITLE
Added test cases for HomeScreenViewModel

### LINKED ISSUE #2599: [Code health] Add test cases for HomeScreenViewModel
Add test cases for HomeScreenViewModel

### PR DESCRIPTION
**Pull request by anandwana001**
Fixes #2599 

@sufyanAbbasi @shobhitagarwal1612  PTAL?

Looks like the codecov is visible after the PR get's merge for outside contributors

Screenshot 2024-08-01 at 12 29 47 PM

### CODE REVIEW

File: ground/src/test/java/com/google/android/ground/ui/home/HomeScreenViewModelTest.kt [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+  @Inject lateinit var navigator: Navigator
+  @Inject lateinit var homeScreenViewModel: HomeScreenViewModel
+  @Inject lateinit var localOfflineAreaStore: LocalOfflineAreaStore
+
+  @Test
+  fun `showAbout is navigating to correct screen`() = runWithTestDispatcher {
+    testNavigateTo(navigator.getNavigateRequests(), HomeScreenFragmentDirections.showAbout()) {
+      homeScreenViewModel.showAbout()
+    }
```
Comment (shobhitagarwal1612) [inline ground/src/test/java/com/google/android/ground/ui/home/HomeScreenViewModelTest.kt:42]: Can we perform UI click instead of calling the nav directions?

**Review by shobhitagarwal1612 — CHANGES_REQUESTED** [review]
Left a comment regarding the testing strategy

File: ground/src/test/java/com/google/android/ground/ui/home/HomeScreenViewModelTest.kt (hunk 1 above)
Comment (anandwana001) [reply ground/src/test/java/com/google/android/ground/ui/home/HomeScreenViewModelTest.kt:42]: ViewModel Testing vs Fragment Testing

Comment (shobhitagarwal1612) [reply ground/src/test/java/com/google/android/ground/ui/home/HomeScreenViewModelTest.kt:42]: @sufyanAbbasi What is your opinion on testing fragments vs view models? I feel testing via fragment gives a better coverage including the UI components. If we add a test for view model right now, that part would still remain missing and having both fragment and view model unit tests would result in duplicate tests.

Comment (anandwana001) [reply ground/src/test/java/com/google/android/ground/ui/home/HomeScreenViewModelTest.kt:42]: Putting my thoughts here:

1. In terms of coverage, yes the above lines are correct
2. Regarding testing strategy, we have two things, for example, A. writing test cases to check Is my view visible on screen or not, B. Are my ViewModel functions working as expected or not, are my ViewModel variables are having correct values or not, no matter how they are consumed by UI or not, but values in variables are correct or not. 
That's why, it's important to have Both the test cases to make sure all pieces of code are working correctly independently.

Comment (shobhitagarwal1612) [reply ground/src/test/java/com/google/android/ground/ui/home/HomeScreenViewModelTest.kt:42]: discussed offline, we'll be going ahead with adding both coverages if possible.

**Review by shobhitagarwal1612 — APPROVED** [review]
